pkg/distributor/distributor_test.go

```diff
@@ -277,6 +321,45 @@ func TestDistributor_Push(t *testing.T) {
 					return testutil.GatherAndCompare(regs[0], strings.NewReader(tc.expectedMetrics), tc.metricNames...)
 				})
 			}
+
+			// Check tracing.
+			span.End()
+			const eventName = "Distributor.prePushValidationMiddleware[deduplicated samples/histograms with conflicting timestamps from write request]"
+			var event trace.Event
+			for _, sp := range spanRecorder.Ended() {
+				if sp.Name() != spanName {
+					continue
+				}
+
+				for _, ev := range sp.Events() {
+					if ev.Name != eventName {
+						continue
+					}
+					require.Emptyf(t, event.Name, "Expected only one event with name %q", eventName)
+
+					event = ev
+				}
+			}
+			if tc.expectDedupTraceEvents {
+				require.NotEmptyf(t, event.Name, "Expected a sample deduplication trace event with name %q", eventName)
+
+				var metric string
+				var count int64
+				for _, attr := range event.Attributes {
+					switch attr.Key {
+					case "metric":
+						metric = attr.Value.AsString()
+					case "count":
+						count = attr.Value.AsInt64()
+					default:
+						require.Failf(t, "Unrecognized event attribute %q", string(attr.Key))
+					}
+				}
+				assert.Equal(t, "test_metric", metric)
+				assert.Equal(t, int64(1), count)
+			} else {
+				require.Emptyf(t, event.Name, "Expected no sample deduplication trace events")
+			}
```

Comment:
Same here, it would be easier to read the test if we had a helper saying `requireContainsAttributes(t, expectedAttributes, event.Attributes)`.